chain/wasm/wasm.go

```diff
@@ -158,6 +160,31 @@ func Prepare(code []byte, params []byte) ([]byte, error) {
 	return parseOutput(instance, ptr.ToI64())
 }
 
+func executeWithTimeout(fn func(...interface{}) (wasm.Value, error), paramsInput int64, wasmInput int64, limitTimeout time.Duration) (wasm.Value, error) {
+	type wasmOutput struct {
+		ptr wasm.Value
+		err error
+	}
+	chanWasmOutput := make(chan wasmOutput, 1)
+
+	go func() {
+		ptr, err := fn(paramsInput, wasmInput)
+		chanWasmOutput <- wasmOutput{ptr: ptr, err: err}
+	}()
+
+	var res wasmOutput
+	select {
+	case <-time.After(limitTimeout):
+		return wasm.Value{}, fmt.Errorf("wasm execution timeout")
+	case res = <-chanWasmOutput:
+		if res.err != nil {
+			return wasm.Value{}, res.err
+		}
+	}
+
+	return res.ptr, res.err
```

Comment:
Do we need to kill the wasm process, or does everything get killed automatically?